src/components/notification/notification.tsx

```diff
@@ -0,0 +1,101 @@
+import React from 'react'
+import classNames from 'classnames'
+import CloseIcon from '../icons/CloseIcon.svg'
+import './notification.less'
+
+type NotificationProps = {
+    id: string
+    icon?: React.ReactNode
+    title?: React.ReactNode
+    subtitle?: React.ReactNode
+    children?: React.ReactNode
+    customCloseButton?: React.ReactNode
+    onClick?: (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void
+    onClose?: (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void
+    closeAltText?: string
+    className?: string
+} & JSX.IntrinsicElements['div']
+
+function Notification({
+    id,
+    icon,
+    title,
+    subtitle,
+    children,
+    customCloseButton,
+    onClick,
+    onClose,
+    closeAltText = 'Close',
+    className,
+    ...rest
+}: NotificationProps) {
+    const titleId = title ? `${id}-title` : null
+    const subtitleId = subtitle ? `${id}-subtitle` : null
+    const contentId = children ? `${id}-content` : null
+    const ariaLabelledBy =
+        children && contentId
```

Comment:
If `children` is true-ish `contentId` is as well, so only checking for one should be enough
```suggestion
        contentId
```

Reply:
Think I originally just checked for children here which made TypeScript angry 😄, but yeah checking for `contentId` is enough 👍